challenges/lib/src/main/java/challenges/tree/Tree.java

```diff
@@ -64,4 +62,22 @@ private void postOrder(Node<T> node, List<T> list) {
         postOrder(node.right, list);
         list.add(node.value);
     }
+
+    public T findMaximumValue() {
+        Queue<Node<T>> queue = new Queue<>();
+        T result = null;
+
+        if (root != null) {
+            queue.enqueue(this.root);
+            result = this.root.value;
+        }
+
+        while (!queue.isEmpty()) {
+            Node<T> node = queue.dequeue();
+            if (node.value.compareTo(result) > 0) result = node.value;
```

Comment:
I've noticed a lot of students doing this on 77. Why not just check if result > node.value? If node.value > result, result == node.value.

Reply:
I can't speak to everyone but I'm using Dynamic types instead of restricting it to an int. This allows for things other than numbers to be stored in my linked list and for the max to still be returned. This was a little bit of a stretch goal and I'm sure there's a better way to do it but it allows for things like Strings to be stored and for it to still return the max.

https://docs.oracle.com/javase/7/docs/api/java/lang/Comparable.html